az-reportal/src/main/java/azkaban/viewer/reportal/ReportalServlet.java

```diff
@@ -1045,7 +1045,19 @@ private String validateAndSaveReport(final HttpServletRequest req,
     page.add("projectId", project.getId());
 
     try {
-      report.createZipAndUpload(projectManager, user, this.reportalStorageUser);
+      // Fetch the uploader's IP
+      String uploaderIPAddr = "";
+      if (req != null) {
+        uploaderIPAddr = req.getHeader("X-FORWARDED-FOR");
+        if (uploaderIPAddr == null || uploaderIPAddr.isEmpty()) {
+          logger.debug("Failed to fetch remote Address using \"X-FORWARDED-FOR\"");
+          uploaderIPAddr = req.getRemoteAddr();
+        }
+        logger.info("uploaderIPAddr = " + uploaderIPAddr);
+      } else {
+        logger.info("HttpServletRequest is NULL");
+      }
+      report.createZipAndUpload(projectManager, user, this.reportalStorageUser, uploaderIPAddr);
```

Comment:
Users are not supposed to upload any files in Reportal instances so this code is never used. I think it's better to assign the default value to the IP address field directly or no value at all and avoid code duplication.

Reply:
I did not know that. If that is the case, perhaps we should remove the whole thing? Lets talk about it offline.